yt/frontends/flash/io.py

```diff
@@ -192,30 +192,35 @@ def _yield_coordinates(self, data_file, needed_ptype=None):
         )
         yield ("io", pxyz)
 
-    def _read_particle_fields(self, chunks, ptf, selector):
-        chunks = list(chunks)
-        data_files = set()
-        assert len(ptf) == 1
-        for chunk in chunks:
-            for obj in chunk.objs:
-                data_files.update(obj.data_files)
+    def _read_data_file(self, data_file, ptf, selector=None):
         px, py, pz = self._position_fields
         p_fields = self._handle["/tracer particles"]
-        for data_file in sorted(data_files, key=lambda x: (x.filename, x.start)):
-            si, ei = data_file.start, data_file.end
-            # This should just be a single item
-            for ptype, field_list in sorted(ptf.items()):
-                x = np.asarray(p_fields[si:ei, px], dtype="=f8")
-                y = np.asarray(p_fields[si:ei, py], dtype="=f8")
-                z = np.asarray(p_fields[si:ei, pz], dtype="=f8")
+        si, ei = data_file.start, data_file.end
```

Comment:
This just makes me realize how clunky this is, when in theory there should be a way to index more directly.